pep-0705.rst

```diff
@@ -0,0 +1,259 @@
+user from passing a ``MovieRecord`` into a ``Movie`` argument in the general case because
+the ``Movie`` class has mutator methods that could potentially allow the function to break
+the type constraints in ``MovieRecord`` by setting ``movie["year"] = None`` or ``del movie["year"]``.
+The problem disappears if we don't have mutator methods in ``Movie``. This could be achieved by defining an immutable interface using a :pep:`544` Protocol::
+
+    from typing import Literal, Protocol, overload
+
+    class Movie(Protocol):
+        @overload
+        def get(self, key: Literal["name"]) -> str: ...
+
+        @overload
+        def get(self, key: Literal["year"]) -> int | None: ...
+
+        @overload
+        def __getitem__(self, key: Literal["name"]) -> str: ...
+
+        @overload
+        def __getitem__(self, key: Literal["year"]) -> int | None: ...
+
+This is very repetitive, easy to get wrong, and is still missing important method definitions like ``__contains__()`` and ``keys()``.
+
+The proposed ``TypedMapping`` type allows a straightforward way of defining these types that should be familiar to existing users of ``TypedDict`` and support the cases exemplified above.
+In addition to those benefits, by flagging arguments of a function as ``TypedMapping``, it makes explicit not just to typecheckers but also to users that the function is not going to modify its inputs, which is usually a desirable property of a function interface.
+Finally, this allows bringing the benefits of ``TypedDict`` to other mapping types that are unrelated to ``dict``.
+
+Specification
+=============
+
+The ``TypedMapping`` type is a protocol behaving almost identically to ``TypedDict``, except:
+
+1. The runtime type of a TypedMapping object is not constrained to be a ``dict``
+2. no mutator methods (``__setitem__``, ``__delitem__``, ``update``, etc.) will be generated
+3. a class definition defines a ``TypedMapping`` protocol if and only if ``TypedMapping`` appears directly in its class bases
+4. subclasses can narrow field types, in the same manner as other protocols
+
+All current and future features of TypedDict are applicable to TypedMapping, including class-based and alternative syntax, totality, and ``Required`` and ``NotRequired`` from :pep:`655`.
+
+As with :pep:`589`, this PEP provides a sketch of how a type checker is expected to support type checking operations involving TypedMapping and TypedDict objects, but details are left to implementors. In particular, type compatibility should be based on structural compatibility.
+
+
+Multiple inheritance and TypedDict
+----------------------------------
+
+A type that inherits from a TypedMapping protocol and from TypedDict (either directly or indirectly):
+
+5. is the structural intersection of its parents, or invalid if no such intersection exists
+6. instances must be a dict subclass
+7. adds mutator methods only for fields it explicitly (re)declares
```

Comment:
```suggestion
* is the structural intersection of its parents, or invalid if no such intersection exists
* instances must be a dict subclass
* adds mutator methods only for fields it explicitly (re)declares
```

I was very confused about the numbering here, as not only is there no unambiguous declared sequence, but it does not appear to be a direct continuation of the previous numbered list, nor was there another obvious reasons for this.